### PR TITLE
solution

### PR DESCRIPTION
**Pull request by winniepuhh**
[DEMO LINK](https://winniepuhh.github.io/react_add-todo-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,129 @@
...
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Post } from './types/types';
+import { TodoList } from './components/TodoList';
+
+export function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const prepearedPosts = todosFromServer.map(post => ({
```
Comment (Vlobdam) [inline src/App.tsx:13]: typo
```suggestion
export const preparedPosts = todosFromServer.map(post => ({
```

**Review by Vlobdam — COMMENTED** [review]
looks good 👍

**Review by Vlobdam — APPROVED** [review]

**Review by StasSokolov1 — CHANGES_REQUESTED** [review]
Almost done, let's improve your code a bit more

File: src/components/UserInfo/UserInfo.tsx [hunk 2]
```diff
@@ -1 +1,19 @@
-export const UserInfo = () => {};
+import { User } from '../../types/types';
+
+interface Props {
+  user: User | null;
```
Comment (BudnikOleksii) [inline src/components/UserInfo/UserInfo.tsx:4]:
```suggestion
  user: User;
```
user should be required, it doesn't make sense to render this component without user

File: src/components/UserInfo/UserInfo.tsx [hunk 3]
```diff
@@ -1 +1,19 @@
...
+}
+
+export const UserInfo: React.FC<Props> = ({ user }) => {
+  if (!user) {
+    return null; // Render nothing if user is null
+  }
+
```
Comment (BudnikOleksii) [inline src/components/UserInfo/UserInfo.tsx:11]:
```suggestion
```

**Review by BudnikOleksii — CHANGES_REQUESTED** [review]
Good job 👍 
Just one moment should be fixed

**Review by BudnikOleksii — APPROVED** [review]
Good work 👍

**Pull request closed without merging**